packages/teleport/src/components/LabelSelector/LabelSelector.tsx

```diff
@@ -0,0 +1,269 @@
+        >
+          <Box
+            bg="#011223"
+            color="white"
+            width={362}
+            p={4}
+            style={{
+              boxShadow: '0px 8px 14px rgba(12, 12, 14, 0.07)',
+              borderRadius: '8px',
+            }}
+          >
+            Teleport provides users the ability to add labels (in the form of
+            key:value pairs) to resources. Some valid example labels are “env:
+            prod” and “arch: x86_64”. Labels, used in conjunction with roles,
+            define access in Teleport. For example, you can specify that users
+            with the “on-call” role can access resources labeled “env: prod”.
+            For more information, check out our documentation on{' '}
+            <a
+              href="https://goteleport.com/docs/setup/admin/trustedclusters/"
+              target="_blank"
+            >
+              RBAC
+            </a>{' '}
+            and{' '}
+            <a
+              href="https://goteleport.com/docs/setup/admin/labels/"
+              target="_blank"
+            >
+              labels
+            </a>
+            .
+          </Box>
+        </Popover>
+        <Text style={{ float: 'right' }}>
+          <a
+            href="https://goteleport.com/docs/setup/admin/labels/"
+            target="_blank"
+            style={{ color: 'rgb(255, 255, 255)' }}
+          >
+            View Documentation
+          </a>
+        </Text>
+      </Heading>
+      <LabelContainer
+        onClick={() => setShowAdd(!showAdd)}
+        data-testid="label-container"
+      >
+        {labels.length === 0 && (
+          <Text style={{ color: 'rgba(255, 255, 255, 0.1)' }}>
+            Click to add new labels.
+          </Text>
+        )}
+        {labelList({ labels, onDismiss: handleLabelDismiss })}
+      </LabelContainer>
+      {showAdd && (
+        <AddLabelContainer
+          ref={addLabelContainerRef}
+          data-testid="add-label-container"
+        >
+          <AddLabelInput
+            name="addLabel"
+            value={newLabel}
+            onChange={e => {
+              setNewLabel(e.target.value);
+            }}
+            onKeyPress={e => {
+              // Add a new label on `Enter` if it's valid.
+              if (e.charCode === 13 && validLabel) {
+                handleAddLabel();
+              }
+            }}
+            ref={addLabelInputRef}
+          />
+          {validLabel ? (
+            <CreateLabel
+              onClick={handleAddLabel}
+              data-testid="create-label-msg"
+            >
+              + Create new label "{newLabel}"
+            </CreateLabel>
+          ) : (
+            <CreateLabelError data-testid="create-label-error">
+              <WarningIconWrapper>
+                <Warning />
+              </WarningIconWrapper>
+              <WarningText>
+                <Text style={{ color: '#D83C31', fontWeight: 700 }}>
+                  Invalid label format
```

Comment:
this error appears before i even write anything, on first click, shouldn't it happen after entering a wrong label?